sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -99,6 +106,71 @@
         }
     }
 
+    /**
+     * Asynchronously acquire a token from Active Directory with Aure CLI.
+     *
+     * @param request the details of the token request
+     * @return a Publisher that emits an AccessToken
+     */
+    public Mono<AccessToken> authenticateWithAzureCli(TokenRequestContext request) {
+        String azCommand = "az account get-access-token --output json --resource ";
+    
+        StringBuilder command = new StringBuilder();
+        command.append(azCommand);
+        String scopes = ScopeUtil.scopesToResource(request.getScopes());
+        command.append(scopes);
+    
+        AccessToken token = null;
+        try {
+            String starter;
+            String switcher;
+            if (System.getProperty("os.name").contains("Windows")) {
+                starter = "cmd.exe";
+                switcher = "/c";
+            } else {
+                starter = "/bin/sh";
+                switcher = "-c";
+            }
+
+            ProcessBuilder builder = new ProcessBuilder(starter, switcher, command.toString());
+            builder.redirectErrorStream(true);
+            Process p = builder.start();
+
+            BufferedReader r = new BufferedReader(new InputStreamReader(p.getInputStream()));
+            String line;
+            StringBuilder output = new StringBuilder();
+            while (true) {
+                line = r.readLine();
+                if (line == null) {
+                    break;
+                }
+                if(line.startsWith("'az' is not recognized") || line.matches("(.*)az:(.*)not found")) {
```

Comment:
Use string constants here.

Reply:
https://github.com/Azure/azure-sdk-for-java/blob/41a932e2bc316f58334cd943499991572c283266/sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java#L495
In the method `openUrl` of class `IdentityClient`, doing similar things with string, didn't use constants. Is it better to follow that way?

Reply:
same as above.